File: src/p0001/python/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        lookup = {}
        for j, y in enumerate(nums):
            x = target - y
            i = lookup.get(x)
            if i is not None:
                result.append(i)
                result.append(j)
                break
            lookup[y] = j
        return result

File: src/p0001/cpp/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> result;
        unordered_map<size_t, size_t> dict;
        for (size_t j = 0; j < nums.size(); j++) {
            size_t y = nums[j], x = target - y;
            auto i = dict.find(x);
            if (i != dict.end()) {
                result.push_back(int(i->second));
                result.push_back(int(j));
                break;
            }
            dict[y] = j;
        }
        return result;
    }
};
